src/parcel_snoopi.jl

```diff
@@ -380,3 +380,150 @@ end
+in the resultant FlameGraph (meaning total time including it and all of its children).
+This can be helpful if you have a very big profile, to save on processing time.
+
+# Examples
+```julia
+julia> timing = SnoopCompileCore.@snoopi_deep begin
+           @eval sort(rand(100))  # Evaluate some code and profile julia's type inference
+       end;
+
+julia> fg = SnoopCompile.to_flamegraph(timing)
+Node(FlameGraphs.NodeData(ROOT() at typeinfer.jl:70, 0x00, 0:15355670))
+
+julia> ProfileView.view(fg);  # Display the FlameGraph in a package that supports it
+
+julia> fg = SnoopCompile.to_flamegraph(timing; tmin_secs=0.0001)  # Skip very tiny frames
+Node(FlameGraphs.NodeData(ROOT() at typeinfer.jl:70, 0x00, 0:15355670))
+```
+
+NOTE: This function must touch every frame in the provided `Timing` to build inclusive
+timing information (`InclusiveTiming`). If you have a very large profile, and you plan to
+call this function multiple times (say with different values for `tmin_secs`), you can save
+some intermediate time by first calling [`build_inclusive_times(t)`](@ref), only once,
+and then passing in the `InclusiveTiming` object for all subsequent calls.
+"""
+function to_flamegraph(t::Timing; tmin_secs = 0.0)
+    it = build_inclusive_times(t)
+    to_flamegraph(it; tmin_secs=tmin_secs)
+end
+
+function to_flamegraph(to::InclusiveTiming; tmin_secs = 0.0)
+    tmin_ns = UInt64(round(tmin_secs * 1e9))
+
+    # Compute a "root" frame for the top-level node, to cover the whole profile
+    node_data = _flamegraph_frame(to, to.start_time; toplevel=true)
+    root = Node(node_data)
+    return _build_flamegraph!(root, to, to.start_time, tmin_ns)
+end
+function _build_flamegraph!(root, to::InclusiveTiming, start_ns, tmin_ns)
+    for child in to.children
+        if child.inclusive_time > tmin_ns
+            node_data = _flamegraph_frame(child, start_ns; toplevel=false)
+            node = addchild(root, node_data)
+            _build_flamegraph!(node, child, start_ns, tmin_ns)
+        end
+    end
+    return root
+end
+
+function frame_name(mi_info::Core.Compiler.Timings.InferenceFrameInfo)
+    frame_name(mi_info.mi::Core.Compiler.MethodInstance)
+end
+function frame_name(mi::Core.Compiler.MethodInstance)
+    frame_name(mi.def.name, mi.specTypes)
+end
+# Special printing for Type Tuples so they're less ugly in the FlameGraph
+function frame_name(name, ::Type{TT}) where TT<:Tuple
+    io = IOBuffer()
+    Base.show_tuple_as_call(io, name, TT)
+    v = String(take!(io))
+    return v
+end
+
+# NOTE: The "root" node doesn't cover th whole profile, because it's only the _complement_
+# of the inference times (so it's missing the _overhead_ from the measurement).
+# SO we need to manually create a root node that covers the whole thing.
+function max_end_time(t::InclusiveTiming)
+    return maximum(child.start_time + child.inclusive_time for child in t.children;
+                    init = UInt64(t.start_time + t.inclusive_time))
```

Comment:
I'm not sure I understand the logic here...why is `start_time` relevant?

Reply:
I've explained why start_time is relevant _in general_ in the review comment above.

In this particular function, it's used to calculate an artificial "<root>" node that extends long enough to cover all of the top-level inference calls. Because without this, I was hitting a problem where the exclusive time of the root node, plus all of the inclusive times of its children, didn't _quite_ cover every nanosecond of the last top-level inference call. I think this is because there's a tiny bit of overhead in the process of starting and stopping the inference which we explicitly exclude from our measurements (in https://github.com/JuliaLang/julia/pull/37749), and that adds up over all the stacks, so that the ROOT() node doesn't quite extend long enough to cover the end time of its last child, and this violates an assertion within `FlameGraphs`.

So this function is used to compute the maximum **end time** by adding each child's start time to its duration. Hopefully that makes sense now?

------

However, after explaining the above, I've now realized that the `children` array is actually sorted by start_time already, since we do depth first search and push_back each node as soon as it's finished. So I guess I could simplify this to simply be
```julia
    last_node = length(t.children) > 0 ? t.children[end] : t
    return last_node.start_time + last_node.inclusive_time
```
Do you think this is clearer?

Reply:
That does seem a bit clearer.

Reply:
Done in latest push. Thanks!